lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -464,19 +480,14 @@ def on_tls_conf_set(
                 callback_override="_restart_opensearch"
             )
 
-    def on_tls_relation_broken(self, event: RelationBrokenEvent):
+    def on_tls_relation_broken(self, _: RelationBrokenEvent):
         """As long as all certificates are produced, we don't do anything."""
         if self._is_tls_fully_configured():
             return
 
         # Otherwise, we block.
         self.unit.status = BlockedStatus(TLSRelationBrokenError)
 
-        try:
-            self._stop_opensearch()
-        except OpenSearchStopError:
-            event.defer()
-
```

Comment:
do we restart OpenSearch without TLS if the relation is broken?

Reply:
We do not. Because if OpenSearch was running, the condition (before this block) `if is_tls_configured` would have been `True`.
As long as the node is well configured and running, we do not break it. New nodes will of course be unable to be configured and will be put on a blocked state.